File: client-jss/src/app/main/services/myjss.category.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from '../../services/appRest.service';
import { MyJssCategory } from '../model/MyJssCategory';

@Injectable({
  providedIn: 'root'
})
export class MyJssCategoryService extends AppRestService<MyJssCategory>{

  constructor(http: HttpClient) {
    super(http, "wordpress");
  }

  getAvailableMyJssCategories() {
    return this.getListCached(new HttpParams(), "myjss-categories");
  }

}

File: src/main/java/com/jss/osiris/modules/myjss/wordpress/service/MyJssCategoryService.java
package com.jss.osiris.modules.myjss.wordpress.service;

import java.util.List;

import com.jss.osiris.modules.myjss.wordpress.model.MyJssCategory;

public interface MyJssCategoryService {
    public MyJssCategory getMyJssCategory(Integer id);

    public List<MyJssCategory> getAvailableMyJssCategories();

    public MyJssCategory addOrUpdateMyJssCategory(MyJssCategory myJssCategory);
}
